pyvelox/pyvelox.h

```diff
@@ -366,11 +400,77 @@ inline void addVectorBindings(
           })
       .def("encoding", &BaseVector::encoding)
       .def("append", [](VectorPtr& u, VectorPtr& v) { appendVectors(u, v); });
-  m.def("from_list", [pool](const py::list& list) mutable {
-    return pyListToVector(list, pool.get());
+  m.def("from_list", [](const py::list& list) mutable {
+    return pyListToVector(list, PyVeloxContext::getInstance().pool());
   });
 }
 
+inline void addExpressionBindings(
+    py::module& m,
+    bool asModuleLocalDefinitions = true) {
+  using namespace facebook::velox;
+  functions::prestosql::registerAllScalarFunctions();
+  parse::registerTypeResolver();
+
+  // PyBind11's classes cannot be const, but the parse functions return const
+  // shared_ptrs, so we wrap in a non-const class.
+  struct IExprWrapper {
+    std::shared_ptr<const core::IExpr> expr;
+  };
+
+  py::class_<IExprWrapper>(
+      m, "Expression", py::module_local(asModuleLocalDefinitions))
+      .def("__str__", [](IExprWrapper& e) { return e.expr->toString(); })
+      .def("getInputs", [](IExprWrapper& e) { return e.expr->getInputs(); })
+      .def(
+          "withInputs",
+          [](IExprWrapper& e, std::vector<IExprWrapper>& i) {
+            std::vector<std::shared_ptr<const core::IExpr>> inputs;
+            inputs.reserve(i.size());
+            for (IExprWrapper& w : i)
+              inputs.push_back(w.expr);
+            IExprWrapper result = {e.expr->withInputs(inputs)};
+            return result;
+          })
+      .def(
+          "evaluate",
+          [](IExprWrapper& e,
+             std::vector<std::string> names,
+             std::vector<VectorPtr>& inputs) {
```

Comment:
Can we have evaluate that takes a map from string-> VectorPtr , so we can have expressions like : 
```
  result =  Expression.from_string("a + b")
                .evaluate({ "a": a, "b":b}) 
  ```

Reply:
Added